rdkit/Chem/ChemUtils/UnitTestSDFToCSV.py

```diff
@@ -40,14 +41,45 @@ def test2(self):
       traceback.print_exc()
       self.fail('conversion failed')
     txt = io.getvalue()
-    lines = txt.split('\n')
-    if not lines[-1]:
-      del lines[-1]
+    lines = [line for line in txt.split('\n') if line.strip() != '']
     self.assertTrue(len(lines) == 6, 'bad num lines: %d' % len(lines))
     line0 = lines[0].split(',')
     self.assertEqual(len(line0), 20)
     self.assertTrue(line0[0] == 'AMW')
     self.assertTrue(line0[1] == 'SMILES')
 
+  def test_parser(self):
+    parser = initParser()
+    # User want's help
+    with self.assertRaises(SystemExit), outputRedirect() as (out, err):
+      parser.parse_args(['-h'])
+    self.assertNotEqual(out.getvalue(), '')
+    self.assertEqual(err.getvalue(), '')
+
+    # Missing input file
+    with self.assertRaises(SystemExit), outputRedirect() as (out, err):
+      parser.parse_args([])
+    self.assertEqual(out.getvalue(), '')
+    self.assertNotEqual(err.getvalue(), '')
+
+    # Input file doesn't exist
+    with self.assertRaises(SystemExit), outputRedirect() as (out, err):
+      parser.parse_args(['incorrectFilename'])
+    self.assertEqual(out.getvalue(), '')
+    self.assertNotEqual(err.getvalue(), '')
+
+
+@contextmanager
```

Comment:
and another construct that I learn in this review. I'm less likely to use this one, but it's still useful